### ISSUE TITLE
Add SecretRequest.spec.fromName to allow renaming on import

### ISSUE THREAD

**Issue by ron1** [carvel-dev/secretgen-controller#7]
**Describe the problem/challenge you have**
I need to rename secret when importing it into a new namespace.

**Describe the solution you'd like**
Add field SecretRequest.spec.fromName which allows SecretRequest.name and SecretRequest.spec.fromName to be different.

**Anything else you would like to add:**
This is needed for scenario where static secrets exist in a common source namespace and have target namespaces embedded in their names. When these secrets are imported into their target namespaces the target names should be stripped.

**Comment by joe-kimmel-vmw** [comment]
@cppforlife any thoughts on tagging this as "accepted" meaning we would merge it if somebody does it?

**Comment by gmrodgers** [comment]
Hey folks, just to follow up, are there concrete things blocking something like this being done? Or is it just waiting for someone to swoop in and do this?

**Comment by neil-hickey** [comment]
I can't see anything being a blocker for this. WDTY @cppforlife ? 

We just don't have the bandwidth to pick it up right now.

**Comment by cppforlife** [comment]
> Or is it just waiting for someone to swoop in and do this?

yup. im also not sure how complicated this would be, so somebody have to do some thinking (in a PR form or comment).
